src/main/java/app/komunumo/ui/views/community/CommunityAddComponent.java

```diff
@@ -0,0 +1,134 @@
+        nameField.setLabel(getTranslation("ui.views.community.AddCommunity.label.name"));
+        nameField.setRequired(true);
+        nameField.setMaxLength(255);
+        nameField.setWidthFull();
+
+        final var descriptionField = new TextArea();
+        descriptionField.setId("description");
+        descriptionField.setValueChangeMode(ValueChangeMode.EAGER);
+        descriptionField.setLabel(getTranslation("ui.views.community.AddCommunity.label.description"));
+        descriptionField.setWidthFull();
+
+        binder.forField(profileField)
+                .asRequired(getTranslation("ui.views.community.AddCommunity.validation.profile.required"))
+                .withValidator(profile -> !communityService.isCommunityExistWithProfile(profile),
+                        getTranslation("ui.views.community.AddCommunity.validation.profile.exists"))
+                .bind(CommunityDto::profile, null);
+
+        binder.forField(nameField)
+                .asRequired(getTranslation("ui.views.community.AddCommunity.validation.name.required"))
+                .bind(CommunityDto::name, null);
+
+        Button saveButton = new Button(getTranslation("ui.views.page.GlobalPageEditorDialog.ConfirmDialog.save"), _ -> {
+            if (binder.validate().isOk()) {
+                var communityDto = new CommunityDto(null, profileField.getValue(), null, null,
+                        nameField.getValue(), descriptionField.getValue(), null);
+
+                communityDto = communityService.storeCommunity(communityDto);
+
+                var memberDto = new MemberDto(SecurityUtil.getUserPrincipal().get().getUserId(),
+                        Objects.requireNonNull(communityDto.id()), "OWNER", null);
+
+                memberService.storeMember(memberDto);
+
+                final var notification = new Notification(getTranslation("ui.views.community.AddCommunity.notification.success"));
+                notification.addThemeVariants(NotificationVariant.LUMO_SUCCESS);
+                notification.setDuration(10_000);
+                notification.open();
+
+                UI.getCurrent().navigate(CommunityGridView.class);
+
+            } else {
+                final var notification = new Notification(getTranslation("ui.views.community.AddCommunity.notification.error"));
+                notification.addThemeVariants(NotificationVariant.LUMO_ERROR);
+                notification.setDuration(10_000);
+                notification.open();
+            }
+        });
+        saveButton.setId("saveButton");
+
+        final var container = new VerticalLayout(profileField, nameField, descriptionField, saveButton);
+        container.addClassName("community-add-component");
+
+        return container;
+    }
+
+    /**
+     * <p>Getter method to retrieve the binder object for this component.</p>
+     *
+     * @return the binder object
+     */
+    public Binder<CommunityDto> getBinder() {
```

Comment:
Looks like you only use it for testing purposes. In that case, please annotate it with `@VisibleForTesting` and make it package visible only.